### PR TITLE
chore(shared-react): move debounce to shared-react

### PR DESCRIPTION
**Pull request by divyanshiGupta**
Fixes: https://github.com/janus-idp/backstage-plugins/issues/582

This PR moves removes duplicate code for debounce and useDeepCompareMemoize files from topology and tekton plugins.

### CODE REVIEW

File: plugins/shared-react/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
   },
   "dependencies": {
     "@kubernetes/client-node": "^0.18.1",
+    "@patternfly/react-icons": "^4.93.6",
```
Comment (debsmita1) [inline plugins/shared-react/package.json:27]: We should avoid creating a dependency on patternfly in this package

**Comment by debsmita1** [comment]
Could you create separate PRs to handle the changes in different plugins ?

File: plugins/shared-react/package.json (hunk 1 above)
Comment (divyanshiGupta) [reply plugins/shared-react/package.json:27]: In that case we might want to keep icons in the plugins itself. Wdyt?  @debsmita1

**Comment by invincibleJai** [comment]
> Could you create separate PRs to handle the changes in different plugins ?

that might not be needed Check this comment from @tumido https://github.com/janus-idp/backstage-plugins/pull/636#discussion_r1297295428

File: plugins/shared-react/package.json (hunk 1 above)
Comment (invincibleJai) [reply plugins/shared-react/package.json:27]: can we use icons from MUI? over pf/react-icons? 

if yes we can keep it here else it's not ideal to bring in pf dependency in a shared package unless it's really needed

Comment (divyanshiGupta) [reply plugins/shared-react/package.json:27]: I will add the similar icons from MUI then.

**Review by invincibleJai — APPROVED** [review]
LGTM

**Comment by tumido** [comment]
@invincibleJai @debsmita1 since this PR was a `chore:`, it did not trigger release of the topology plugin. Please mind that this change will be introduced in the next release (with next change to topology plugin).

**Pull request merged**